packages/api/src/modules/accounts/whiteList.service.ts

```diff
@@ -89,20 +89,20 @@ export class KycService {
     if (existing.data) return { data: [existing.data] };
 
     return admin
-      .from('kyc_events')
+      .from('whitelist_events')
       .insert({ ...dto, event_name: 'accepted' })
       .select();
   }
 
   private async checkOnChain(
-    kycProviders: Tables<'vault_entities'>[],
+    whiteListProviders: Tables<'vault_entities'>[],
     address: string,
   ): Promise<ServiceResponse<boolean>> {
     const errors = [];
-    let status = false;
```

Comment:
This starting value of `false` means that the below loop will never return a true. This is a bug, I believe and has been fixed.

Reply:
Confirming this is a bug!

For the fix, On line 113, the NOT(!) operator should be removed.
The loop should break if the status is once set to true after the line `status = status && data`

Reply:
So you are saying ANY White List Provider can whitelist an address?
Actually, it sounds logical when you put it that way. Thanks, will fix.